core/src/main/java/aima/core/environment/map2d/Print2DMap.java

```diff
@@ -0,0 +1,125 @@
+package aima.core.environment.map2d;
+
+import aima.core.util.datastructure.Point2D;
+
+import javax.imageio.ImageIO;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+
+/**
+ * I was sceptic if the java implementation matches the images I found. Now one can compare.
+ * @author wormi
+ */
+public class Print2DMap {
+
+  private static double width;
+  private static double height;
+  private static double zoom = 1;
+  private static double offsetX;
+  private static double offsetY;
+  // some border is needed because text does not stay within width
+  private static int border = 300;
+
+  public static void main (String[] args) throws URISyntaxException, IOException, ClassNotFoundException, IllegalAccessException, InstantiationException {
+    if (args.length < 2) {
+      System.err.println("run Print2DMap through your IDE," +
+          " `Map2DClassSimpleName yourOutPath [zoom] [border]` as arguments\n" +
+          " e.g `SimplifiedRoadMapOfPartOfRomania /tmp 1.4 100`\n" +
+          "The map to print must implement the `Map2D` interface\n" +
+          "and must reside in the same package");
+      System.exit(1);
+    }
+    if (args.length >= 3) {
+      zoom = Double.parseDouble(args[2]);
+    }
+    if (args.length >= 4) {
+      border = Integer.parseInt(args[3]);
+    }
+
+    final String fullQualifiedClassName = Print2DMap.class.getPackage().getName() + "." + args[0];
+    Class c= Class.forName(fullQualifiedClassName);
+    Map2D mapFromClassPath = (Map2D) c.newInstance();
+
+    calculateDimensions(mapFromClassPath);
+
+    int _width = ((Double) width).intValue();
+    int _height = ((Double) height).intValue();
+    BufferedImage image = new BufferedImage(_width, _height, BufferedImage.TYPE_INT_RGB);
+    final Graphics graphics = image.createGraphics();
+    graphics.setColor(Color.WHITE);
+    graphics.fillRect(0, 0, _width, _height);
+    draw(graphics, mapFromClassPath);
+
+    String path = args[1];
+    if (path.endsWith("/")) {
```

Comment:
And what about Windows! ;) You can simply create a file object and ask if its a directory or not.